spark-atlas-connector/src/main/scala/com/hortonworks/spark/atlas/types/external.scala

```diff
@@ -108,23 +108,32 @@ object external {
     }
   }
 
+  private def qualifiedPath(path: String): Path = {
+    val p = new Path(path)
+    val fs = p.getFileSystem(SparkUtils.sparkSession.sparkContext.hadoopConfiguration)
+    p.makeQualified(fs.getUri, fs.getWorkingDirectory)
+  }
+
   private def resolveURI(path: String): URI = {
-    try {
-      val uri = new URI(path)
-      if (uri.getScheme() != null) {
-        return uri
-      }
+    val uri = new URI(path)
+    if (uri.getScheme() != null) {
+      return uri
+    }
+
+    val qUri = qualifiedPath(path).toUri
```

Comment:
If its the case that all paths in spark has to be a Hadoop Path then this logic seems OK.
If its possible that in some source/sink a path may not be a Hadoop path (e.g. a java.nio.Path referring to a local file and scheme not specified), but we resolve it as HDFS path when it reaches here (because fs.Defaultfs is hdfs://...) ?

Reply:
Unless Spark puts extra efforts to take care of both file systems (which I don't think Spark would), Spark also has no idea to determine whether the given path is local or HDFS, and Spark has to depend on `fs.defaultFS`.

Other than that, this patch doesn't break existing UTs, hence the patch guarantees working with the case where path is in file system and scheme is not specified, and fs.defaultFS is not set to `hdfs://`.